src/api/v1/serializers/pizzas-serializer.js

```diff
@@ -0,0 +1,107 @@
+import { Serializer as JsonApiSerializer } from 'jsonapi-serializer';
+import _ from 'lodash';
+
+import { transformDough } from 'api/v1/serializers/doughs-serializer';
+import { transformIngredient } from 'api/v1/serializers/ingredients-serializer';
+import { serializerOptions } from 'utils/jsonapi';
+import { openapi } from 'utils/load-openapi';
+import { apiBaseUrl, resourcePathLink } from 'utils/uri-builder';
+
+
+const pizzaResourceProp = openapi.definitions.PizzaResource.properties;
+const pizzaResourceType = pizzaResourceProp.type.enum[0];
+const pizzaResourceKeys = _.keys(pizzaResourceProp.attributes.properties);
+const pizzaResourcePath = 'pizzas';
+
+const doughResourceProp = openapi.definitions.DoughRecipe.properties;
+const doughResourceKeys = _.keys(doughResourceProp.attributes.properties);
+
+const ingredientResourceProp = openapi.definitions.IngredientResource.properties;
+const ingredientResourceKeys = _.keys(ingredientResourceProp.attributes.properties);
+
+/**
+ * Replace `null` with empty string and conver to int as appropriate
+ *
+ * @param {object} rawPizza a pizza object from the database
+ * @returns {object} the transformed pizza
+ */
+const transformRawPizza = (rawPizza) => {
+  rawPizza.ingredients.map((rawIngredient) => transformIngredient(rawIngredient));
+  rawPizza.dough = transformDough(rawPizza.dough);
+
+  rawPizza.specialInstructions = rawPizza.specialInstructions || '';
+
+  [
+    'ovenTemp',
+    'bakeTime',
+  ].forEach((integerField) => {
+    rawPizza[integerField] = parseInt(rawPizza[integerField], 10);
+  });
+
+  return rawPizza;
+};
+
+/**
+ * Add a compound relationship named `relationName` to `options`
+ * with keys `relationKeys` and primary key field `idField`
+ *
+ * @param {object} options
+ * @param {string} relationName
+ * @param {object} relationKeys
+ * @param {string} idField
+ * @returns {object} `options` with `relationName` added
+ */
+const addCompoundRelationship = (options, relationName, relationKeys, idField = 'id') => {
```

Comment:
I might miss it, but will `idField` be overwritten by some other fields?

Reply:
what do you mean?

Reply:
So, from what I can see so far, this function was called in two places:

```js
options = addCompoundRelationship(options, 'dough', doughResourceKeys);
options = addCompoundRelationship(options, 'ingredients', ingredientResourceKeys);
```

My question was that since`idField` was not even get passed when you call this function, do we need to make it as a parameter?